tests/test_visitors/test_ast/test_statements/test_variable_used_outside_of_block.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+
+from wemake_python_styleguide.violations.best_practices import (
+    VariableUsedOutsideOfForBlockViolation,
+    VariableUsedOutsideOfWithBlockViolation
+)
+from wemake_python_styleguide.visitors.ast.statements import (
+    VariableUsedOutsideOfBlockVisitor,
+)
+
+variable_used_outside_with = """
+def function():
+    with a.open() as x:
+        some()
+        function()
+        calls()
+    print("This is not allowed".format(x))
+"""
+multiple_variables_defined_by_with = """
```

Comment:
New line is required after each multiline example. I will make it a linting rule! Thanks for bringing this up!